sqllogictest/src/runner.rs

```diff
@@ -854,3 +855,174 @@ impl<D: AsyncDB> Runner<D> {
 fn normalize_string(s: &String) -> String {
     s.trim().split_ascii_whitespace().join(" ")
 }
+
+/// Updates the specified [`Record`] with the [`QueryOutput`] produced
+/// by a Database, returning `Some(new_record)`.
+///
+/// If an update is not supported, returns `None`
+pub fn update_record_with_output(record: &Record, record_output: &RecordOutput) -> Option<Record> {
+    match (record, record_output) {
+        (
+            Record::Query {
+                loc,
+                conditions,
+                type_string,
+                sort_mode,
+                label,
+                expected_error,
+                sql,
+                expected_results: _,
+            },
+            RecordOutput::Query {
+                types: _,
+                rows,
+                error,
+            },
+        ) if error.is_none() && expected_error.is_none() => {
+            let expected_results = rows.iter().map(|cols| cols.join(" ")).collect();
+
+            Some(Record::Query {
+                loc: loc.clone(),
+                conditions: conditions.clone(),
+                type_string: type_string.clone(),
+                sort_mode: sort_mode.clone(),
+                label: label.clone(),
+                expected_error: expected_error.clone(),
+                sql: sql.clone(),
+                expected_results,
+            })
+        }
+
+        // No update possible, return the original record
```

Comment:
Did you look at `update_record`? https://github.com/risinglightdb/sqllogictest-rs/blob/46752d1b536f186e3f976bdfce4968ce761ab230/sqllogictest-bin/src/lib.rs

I think they are highly alike, but it contains more updatable edge cases. How do you think the differences?

Reply:
🤔  I forgot out `update_record` 

I think it is cleaner to update the `Record` and then use the `Display` logic to print it back out, rather than write to a new file directly. However, there is nothing wrong with generating the text directly (but there may be more chance of inconsistencies)

I am out of time for coding today, sadly, but I will think on this question and see if I can come to a conclusion

Reply:
@xxchan if I want to refactor `update_record` and avoid breaking your tests, what is the best way to do so? I have been running `cargo test` but I guess that is not the only thing 🤔

Reply:
@alamb I did the refactor. You can take a look whether it looks good to you.

Reply:
I took a look - and it looks good to me @xxchan -- thank you 🙏 

The only thing I would normally do is to write some more tests to cover these cases (and also illustrate how the updates work)-- perhaps I can do that as a follow on PR?